src/include/mpir_cclcomm.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef MPIR_CCLCOMM_H_INCLUDED
+#define MPIR_CCLCOMM_H_INCLUDED
+
+#include <nccl.h>
+
+typedef struct MPIR_CCLcomm {
+    MPIR_OBJECT_HEADER;
+    MPIR_Comm *comm;
+    ncclUniqueId id;
+    ncclComm_t ncclcomm;
+    cudaStream_t stream;
+} MPIR_CCLcomm;
```

Comment:
will need the right autoconf/automake defines to only include this stuff when its available.